pandas/tests/indexes/test_multi.py

```diff
@@ -2629,6 +2629,16 @@ def test_reconstruct_remove_unused(self):
         tm.assert_index_equal(result2, expected)
         assert result2.is_(result)
 
+    @pytest.mark.parametrize('level', [['a', 'd', 'b'],
+                                       ['a', 'd', 'b', 'unused']])
+    def test_remove_unused_nan(self, level):
+        # GH 18417
+        df = pd.MultiIndex(levels=[level, ['w', 'x', 'y', 'z']],
+                           labels=[[0, 2, -1, 1, 1], [0, 1, 2, 3, 2]])
+
+        result = df.remove_unused_levels()
+        tm.assert_index_equal(result, df)
```

Comment:
Seems not drop `'unused'` when `level = ['a', 'd', 'b', 'unused']`.

Reply:
It does... do you mean "you should check it does"?

Reply:
@toobaz I am asking why original `df` and dropped `result` are same.

Reply:
In `level = ['a', 'd', 'b', 'unused']` case, `'unused'` is unused and should be dropped.

Reply:
> @toobaz I am asking why original df and dropped result are same.

They are equivalent precisely because the dropped level is not used. "[The resulting MultiIndex will have the same outward appearance, meaning the same .values and ordering. It will also be .equals() to the original.](https://pandas.pydata.org/pandas-docs/stable/generated/pandas.MultiIndex.remove_unused_levels.html)"

Reply:
> 'unused' is unused and should be dropped.

This is exactly what happens.

Reply:
@toobaz Okay, you are right.
But you have to check whether `unused` is really dropped.